app/views/topics/show.html.erb

```diff
@@ -1,14 +1,25 @@
-<h1><%= @topic.title %></h1>
+<div class="row">
+  <div class="col-md-3">
+    <h1><%= @topic.title %></h1>
+  </div>
+  <div class="col-md-3">
+    <h1><%= link_to "Edit Topic", edit_topic_path, class: "btn btn-block btn-lg btn-warning" %></h1>
+  </div>
+  <div class="col-md-3">
+    <h1><%= link_to "Delete Topic", @topic, method: :delete, class: "btn btn-block btn-lg btn-danger" %></h1>
+  </div>
+  <div class="col-md-3">
+    <h1><%= link_to "New Bookmark", new_topic_bookmark_path(@topic, @post), class: "btn btn-block btn-lg btn-primary" %></h1>
+  </div>
+</div>
 <div class="row">
   <div class="col-md-8">
-    <% @topic_bookmarks.each do |topic_bookmark| %>
+    <% @bookmarks.each do |topic_bookmark| %>
       <% @url = "#{topic_bookmark.url}" %>
```

Comment:
I am iterating over all the bookmark urls for that given topic. Also, I removed the string interpolation, I changed it from @url = "#{topic_bookmark.url}" to: url = topic_bookmark.url  . I needed the url for the href  and for the text shown with in the link < a href= <%= url %> >  <%= url %>  </a>